### ISSUE TITLE
[a11y]: Datatable expand failed for accessibility check

### ISSUE THREAD

**Issue by tummyyu** [carbon-design-system/carbon#19133]
### Package

@carbon/react

### Browser

Firefox

### Operating System

MacOS

### Package version

11.23.1

### React version

1.71.0

### Automated testing tool and ruleset

IBM Equal Access Accessibility Checker

### Description

Data table failed for expand feature enabled.
And tools report 

for batch expand button:

The 'id' "expanded-row-0, expanded-row-1, expanded-row-2, expanded-row-3, expanded-row-4, expanded-row-5, expanded-row-6, expanded-row-7, expanded-row-8, expanded-row-9, expanded-row-10, expanded-row-11, expanded-row-12, expanded-row-13, expanded-row-14, expanded-row-15, expanded-row-16, expanded-row-17, expanded-row-18, expanded-row-19, expanded-row-20, expanded-row-21, expanded-row-22, expanded-row-23, expanded-row-24, expanded-row-25, expanded-row-26, expanded-row-27" specified for the ARIA property 'aria-controls' value is not valid

For single expand button:
The 'id' "expanded-row-0" specified for the ARIA property 'aria-controls' value is not valid

Looks aria-controls didn't find corresponded id. 

### Reproduction/example

no

### Steps to reproduce

Just use example from here https://react.carbondesignsystem.com/?path=/story/components-datatable-expansion--batch-expansion

### Suggested Severity

None

### Code of Conduct

- [x] I agree to follow this project's [Code of Conduct](https://github.com/carbon-design-system/carbon/blob/f555616971a03fd454c0f4daea184adf41fff05b/.github/CODE_OF_CONDUCT.md)
- [x] I checked the [current issues](https://github.com/carbon-design-system/carbon/issues) for duplicate problems
